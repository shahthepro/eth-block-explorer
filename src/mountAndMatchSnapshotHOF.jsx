export default function (it, ReactDOM, mount) {
  return function (elementName, element) {
    it(`renders <${elementName}> without crashing`, () => {
      const div = document.createElement('div');
      ReactDOM.render(element, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    
    it(`matches <${elementName}> snapshot`, () => {
      const mountedElement = mount(element);
      expect(mountedElement).toMatchSnapshot();
    });
  }
}