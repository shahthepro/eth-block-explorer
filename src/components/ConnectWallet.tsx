import * as React from 'react';
import * as PropTypes from 'prop-types';
import styled from 'styled-components';
import { IWalletState } from "src/types";
import { Title, Subtitle } from 'src/components/styles/Typography';
import { PrimaryButton } from 'src/components/styles/Buttons';
import { ErrorMessage } from './styles/ErrorStyles';
// import getNetworkName from 'src/lib/getNetworkName';

interface IConnectWalletProps {
  wallet: IWalletState
  children: any
}

const ConnectWalletWrapper = styled.div`
  padding: 2rem;
  border: 1px solid #f0f0f0;
  box-sizing: border-box;
  height: 100%;
  overflow: auto;
`;

const ConnectWallet = ({ wallet, children }: IConnectWalletProps) => {
  if (wallet.isConnected) {
    return children;
  }

  return <ConnectWalletWrapper>
    <Title>Hey there!</Title>
    <Subtitle>Connect your software wallet to explore recent blocks</Subtitle>
    {
      wallet.lastConnectionError &&
      <ErrorMessage>{ wallet.lastConnectionError }</ErrorMessage>
    }
    <PrimaryButton onClick={wallet.connect} disabled={wallet.isConnected || wallet.isConnecting}>Connect Wallet</PrimaryButton>
  </ConnectWalletWrapper>
};

ConnectWallet.propTypes = {
  wallet: PropTypes.object.isRequired,
};

export default ConnectWallet;