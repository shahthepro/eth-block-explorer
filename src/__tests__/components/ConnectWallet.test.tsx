import * as React from 'react';
import * as ReactDOM from 'react-dom';
import ConnectWallet from './../../components/ConnectWallet';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mountAndMatchSnapshotHOF from '../../mountAndMatchSnapshotHOF';

Enzyme.configure({ adapter: new Adapter() })
const { mount } = Enzyme;

const mountAndMatchSnapshot = mountAndMatchSnapshotHOF(it, ReactDOM, mount);

// @ts-ignore
mountAndMatchSnapshot('ConnectWallet wallet.isConnected', <ConnectWallet wallet={{
  isConnected: true
}}><div>child</div></ConnectWallet>);

// @ts-ignore
mountAndMatchSnapshot('ConnectWallet wallet.isConnected={false}', <ConnectWallet wallet={{
  isConnected: false,
  connect: () => {},
  isConnecting: false,
  lastConnectionError: ''
}}><div>child</div></ConnectWallet>);

// @ts-ignore
mountAndMatchSnapshot('ConnectWallet wallet.isConnecting={true}', <ConnectWallet wallet={{
  isConnected: false,
  connect: () => {},
  isConnecting: true,
  lastConnectionError: ''
}}><div>child</div></ConnectWallet>);

// @ts-ignore
mountAndMatchSnapshot('ConnectWallet wallet.lastConnectionError={"error"}', <ConnectWallet wallet={{
  isConnected: false,
  connect: () => {},
  isConnecting: true,
  lastConnectionError: 'error'
}}><div>child</div></ConnectWallet>);