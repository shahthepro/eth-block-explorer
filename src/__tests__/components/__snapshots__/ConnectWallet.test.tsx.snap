// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches <ConnectWallet wallet.isConnected={false}> snapshot 1`] = `
<ConnectWallet
  wallet={
    Object {
      "connect": [Function],
      "isConnected": false,
      "isConnecting": false,
      "lastConnectionError": "",
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-dnqmqq",
            "isStatic": true,
            "lastClassName": "jnlZaT",
            "rules": Array [
              "
          padding: 2rem;
          border: 1px solid #f0f0f0;
          box-sizing: border-box;
          height: 100%;
          overflow: auto;
        ",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-dnqmqq",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-dnqmqq jnlZaT"
      >
        <styled.h1>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "gkfbEN",
                  "rules": Array [
                    "
                font-weight: 100;
                margin: 0;
                padding-bottom: 1rem;
                text-align: ",
                    [Function],
                    ";
              ",
                  ],
                },
                "displayName": "styled.h1",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "h1",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h1
              className="sc-bdVaJa gkfbEN"
            >
              Hey there!
            </h1>
          </StyledComponent>
        </styled.h1>
        <styled.h2>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "QULuE",
                  "rules": Array [
                    "
                font-weight: 100;
                margin: 0;
                padding-bottom: 1rem;
              ",
                  ],
                },
                "displayName": "styled.h2",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "h2",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h2
              className="sc-bwzfXH QULuE"
            >
              Connect your software wallet to explore recent blocks
            </h2>
          </StyledComponent>
        </styled.h2>
        <Styled(styled.button)
          disabled={false}
          onClick={[Function]}
        >
          <StyledComponent
            disabled={false}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": false,
                  "lastClassName": "cOXKDn",
                  "rules": Array [
                    "
                background-color: rgba(0, 122, 255, 1);
                color: #fff;
                border: 0;
                padding: ",
                    [Function],
                    ";
                font-size: ",
                    [Function],
                    ";
                cursor: pointer;
              
                &:hover {
                  background-color: rgba(0, 122, 255, 0.8);
                }
              
                &:disabled {
                  background-color: #ccc;
                  color: #888;
                }
              ",
                    "
                background-color: rgba(0, 122, 255, 1);
              
                &:hover {
                  background-color: rgba(0, 122, 255, 0.8);
                }
              ",
                  ],
                },
                "displayName": "Styled(styled.button)",
                "foldedComponentIds": Array [
                  "sc-htpNat",
                ],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="sc-htpNat sc-bxivhb cOXKDn"
              disabled={false}
              onClick={[Function]}
            >
              Connect Wallet
            </button>
          </StyledComponent>
        </Styled(styled.button)>
      </div>
    </StyledComponent>
  </styled.div>
</ConnectWallet>
`;

exports[`matches <ConnectWallet wallet.isConnected> snapshot 1`] = `
<ConnectWallet
  wallet={
    Object {
      "isConnected": true,
    }
  }
>
  <div>
    child
  </div>
</ConnectWallet>
`;

exports[`matches <ConnectWallet wallet.isConnecting={true}> snapshot 1`] = `
<ConnectWallet
  wallet={
    Object {
      "connect": [Function],
      "isConnected": false,
      "isConnecting": true,
      "lastConnectionError": "",
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-dnqmqq",
            "isStatic": true,
            "lastClassName": "jnlZaT",
            "rules": Array [
              "
          padding: 2rem;
          border: 1px solid #f0f0f0;
          box-sizing: border-box;
          height: 100%;
          overflow: auto;
        ",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-dnqmqq",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-dnqmqq jnlZaT"
      >
        <styled.h1>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "gkfbEN",
                  "rules": Array [
                    "
                font-weight: 100;
                margin: 0;
                padding-bottom: 1rem;
                text-align: ",
                    [Function],
                    ";
              ",
                  ],
                },
                "displayName": "styled.h1",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "h1",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h1
              className="sc-bdVaJa gkfbEN"
            >
              Hey there!
            </h1>
          </StyledComponent>
        </styled.h1>
        <styled.h2>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "QULuE",
                  "rules": Array [
                    "
                font-weight: 100;
                margin: 0;
                padding-bottom: 1rem;
              ",
                  ],
                },
                "displayName": "styled.h2",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "h2",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h2
              className="sc-bwzfXH QULuE"
            >
              Connect your software wallet to explore recent blocks
            </h2>
          </StyledComponent>
        </styled.h2>
        <Styled(styled.button)
          disabled={true}
          onClick={[Function]}
        >
          <StyledComponent
            disabled={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": false,
                  "lastClassName": "cOXKDn",
                  "rules": Array [
                    "
                background-color: rgba(0, 122, 255, 1);
                color: #fff;
                border: 0;
                padding: ",
                    [Function],
                    ";
                font-size: ",
                    [Function],
                    ";
                cursor: pointer;
              
                &:hover {
                  background-color: rgba(0, 122, 255, 0.8);
                }
              
                &:disabled {
                  background-color: #ccc;
                  color: #888;
                }
              ",
                    "
                background-color: rgba(0, 122, 255, 1);
              
                &:hover {
                  background-color: rgba(0, 122, 255, 0.8);
                }
              ",
                  ],
                },
                "displayName": "Styled(styled.button)",
                "foldedComponentIds": Array [
                  "sc-htpNat",
                ],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="sc-htpNat sc-bxivhb cOXKDn"
              disabled={true}
              onClick={[Function]}
            >
              Connect Wallet
            </button>
          </StyledComponent>
        </Styled(styled.button)>
      </div>
    </StyledComponent>
  </styled.div>
</ConnectWallet>
`;

exports[`matches <ConnectWallet wallet.lastConnectionError={"error"}> snapshot 1`] = `
<ConnectWallet
  wallet={
    Object {
      "connect": [Function],
      "isConnected": false,
      "isConnecting": true,
      "lastConnectionError": "error",
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-dnqmqq",
            "isStatic": true,
            "lastClassName": "jnlZaT",
            "rules": Array [
              "
          padding: 2rem;
          border: 1px solid #f0f0f0;
          box-sizing: border-box;
          height: 100%;
          overflow: auto;
        ",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-dnqmqq",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-dnqmqq jnlZaT"
      >
        <styled.h1>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "gkfbEN",
                  "rules": Array [
                    "
                font-weight: 100;
                margin: 0;
                padding-bottom: 1rem;
                text-align: ",
                    [Function],
                    ";
              ",
                  ],
                },
                "displayName": "styled.h1",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "h1",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h1
              className="sc-bdVaJa gkfbEN"
            >
              Hey there!
            </h1>
          </StyledComponent>
        </styled.h1>
        <styled.h2>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "QULuE",
                  "rules": Array [
                    "
                font-weight: 100;
                margin: 0;
                padding-bottom: 1rem;
              ",
                  ],
                },
                "displayName": "styled.h2",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "h2",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h2
              className="sc-bwzfXH QULuE"
            >
              Connect your software wallet to explore recent blocks
            </h2>
          </StyledComponent>
        </styled.h2>
        <Styled(styled.div)>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htoDjs",
                  "isStatic": true,
                  "lastClassName": "fknPkg",
                  "rules": Array [
                    "
                padding: 1rem;
                background: white;
                margin: 1rem 0;
                border: 1px solid rgba(0, 0, 0, 0.05);
                border-left: 5px solid red;
                p {
                  margin: 0;
                  font-weight: 100;
                }
                strong {
                  margin-right: 1rem;
                }
              ",
                    "
                border-left: 5px solid red;
              ",
                  ],
                },
                "displayName": "Styled(styled.div)",
                "foldedComponentIds": Array [
                  "sc-bZQynM",
                ],
                "render": [Function],
                "styledComponentId": "sc-htoDjs",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bZQynM sc-htoDjs fknPkg"
            >
              error
            </div>
          </StyledComponent>
        </Styled(styled.div)>
        <Styled(styled.button)
          disabled={true}
          onClick={[Function]}
        >
          <StyledComponent
            disabled={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": false,
                  "lastClassName": "cOXKDn",
                  "rules": Array [
                    "
                background-color: rgba(0, 122, 255, 1);
                color: #fff;
                border: 0;
                padding: ",
                    [Function],
                    ";
                font-size: ",
                    [Function],
                    ";
                cursor: pointer;
              
                &:hover {
                  background-color: rgba(0, 122, 255, 0.8);
                }
              
                &:disabled {
                  background-color: #ccc;
                  color: #888;
                }
              ",
                    "
                background-color: rgba(0, 122, 255, 1);
              
                &:hover {
                  background-color: rgba(0, 122, 255, 0.8);
                }
              ",
                  ],
                },
                "displayName": "Styled(styled.button)",
                "foldedComponentIds": Array [
                  "sc-htpNat",
                ],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="sc-htpNat sc-bxivhb cOXKDn"
              disabled={true}
              onClick={[Function]}
            >
              Connect Wallet
            </button>
          </StyledComponent>
        </Styled(styled.button)>
      </div>
    </StyledComponent>
  </styled.div>
</ConnectWallet>
`;
